errors.go

```diff
@@ -122,6 +122,10 @@ func (mc *mysqlConn) getWarnings() (err error) {
 			warnings = append(warnings, warning)
 
 		case io.EOF:
+			if len(warnings) == 0 {
+				return nil
```

Comment:
Shouldn't this be rather [`fallthrough`](https://github.com/golang/go/wiki/Switch#fall-through) (to `default`)?